pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -783,19 +783,23 @@ func ApplyRingHashLoadBalancer(c *cluster.Cluster, lb *networking.LoadBalancerSe
 	if consistentHash == nil {
 		return
 	}
-
-	// TODO MinimumRingSize is an int, and zero could potentially be a valid value
-	// unable to distinguish between set and unset case currently GregHanson
-	// 1024 is the default value for envoy
-	minRingSize := &wrappers.UInt64Value{Value: 1024}
-	if consistentHash.MinimumRingSize != 0 {
-		minRingSize = &wrappers.UInt64Value{Value: consistentHash.GetMinimumRingSize()}
-	}
-	c.LbPolicy = cluster.Cluster_RING_HASH
-	c.LbConfig = &cluster.Cluster_RingHashLbConfig_{
-		RingHashLbConfig: &cluster.Cluster_RingHashLbConfig{
-			MinimumRingSize: minRingSize,
-		},
+	if features.UseMaglevForConsistentHash {
+		// Maglev performs consistently better when consistent hashing is needed.
+		c.LbPolicy = cluster.Cluster_MAGLEV
+	} else {
+		// TODO MinimumRingSize is an int, and zero could potentially be a valid value
+		// unable to distinguish between set and unset case currently.
+		// 1024 is the default value for envoy
+		minRingSize := &wrappers.UInt64Value{Value: 1024}
+		if consistentHash.MinimumRingSize != 0 {
```

Comment:
Now this is ignored. We may want to warn if its set? Or possibly even switch to ring_hash if its set?

Reply:
warning seems better and if user wants RING_HASH he can flip the flag. Should we deprecate this if we moved forward with this?